pkg/envinfo/envinfo.go

```diff
@@ -301,6 +301,17 @@ func (esi *EnvSpecificInfo) DeleteURL(parameter string) error {
 	return esi.writeToFile()
 }
 
+func (esi *EnvSpecificInfo) DeleteLink(parameter string) error {
+	for i, link := range *esi.componentSettings.Link {
```

Comment:
you shouldn't change the `Link` while you are looping though it. better just do 
```
index := -1
...
if link.Name == parameter {
   index = i
   break
}
...
if index != -1 {
          s := *esi.componentSettings.Link
          s = append(s[:index], s[index+1:]...)
	  esi.componentSettings.Link = &s
}
```

Reply:
Doesn't `*esi.componentSettings.Link` need to be checked to make sure it's not == `nil` as well? Looks like a null pointer error waiting to happen.

Reply:
@cdrage there is a check to validate that a link exists before we reach this function. Here's how it looks when executed:
```sh
$ cat .odo/env/env.yaml
ComponentSettings:
  Name: node-todo
  Namespace: myproject
  AppName: app
  RunMode: run

$ odo unlink EtcdCluster/example
 ✗  failed to unlink the service "EtcdCluster/example" since it's not linked with the component "node-todo"
```